src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -62,6 +62,8 @@ public class RestoreArgs
 
         public Guid ParentId { get; set; }
 
+        public bool IsRestore { get; set; } = true;
```

Comment:
bwcause `RestoreArgs` is also used to run install/ update for build integrated projects. And we need a way to differentiate between restore and install so that when updating lock file on restore is disabled then we can throw appropriate error which is only applicable for "restore".

I'm open for suggestion of better names.